sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,574 @@
+            AddFunctionMetadataInfo(indentedTextWriter, receiver, compilation);
+            indentedTextWriter.WriteLine("return Task.FromResult(metadataList.ToImmutableArray());");
+
+            indentedTextWriter.Indent--;
+            indentedTextWriter.WriteLine("}");
+        }
+
+        private static void AddFunctionMetadataInfo(IndentedTextWriter indentedTextWriter, SyntaxReceiver receiver, Compilation compilation)
+        {
+            var assemblyName = compilation.Assembly.Name;
+            var scriptFile = Path.Combine(assemblyName + ".dll");
+
+            // Loop through the candidate methods (these are mthods suspected to be Functions, stored in the SyntaxReceiver)
+            foreach (MethodDeclarationSyntax method in receiver.CandidateMethods)
+            {
+                var functionClass = (ClassDeclarationSyntax)method.Parent!;
+                var functionName = functionClass.Identifier.ValueText;
+                var entryPoint = assemblyName + "." + functionName + "." + method.Identifier.ValueText;
+                var bindingsListName = functionName + "RawBindings";
+
+                // collect Bindings
+                indentedTextWriter.WriteLine($"var {bindingsListName} = new List<string>();");
+                AddBindingInfo(indentedTextWriter, method, compilation, functionName);
+                indentedTextWriter.WriteLine($"var {functionName} = new DefaultFunctionMetadata(Guid.NewGuid().ToString(), \"dotnet-isolated\", \"{functionName}\", \"{entryPoint}\", {functionName}RawBindings, \"{scriptFile}\");");
+                indentedTextWriter.WriteLine($"metadataList.Add({functionName});");
+            }
+        }
+
+        private static void AddBindingInfo(IndentedTextWriter indentedTextWriter, MethodDeclarationSyntax method, Compilation compilation, string functionName)
+        {
+            var model = compilation.GetSemanticModel(method.SyntaxTree);
+            var bindingCount = 0;
+            var bindingsListName = functionName + "RawBindings";
+
+            // Loop through the parameters in this Function to find the bindings
+            foreach (ParameterSyntax parameter in method.ParameterList.Parameters)
+            {
+                // If there's no attribute, we can assume that this parameter is not a binding
+                if (parameter.AttributeLists.Count == 0)
+                {
+                    continue;
+                }
+
+                IParameterSymbol? parameterSymbol = model.GetDeclaredSymbol(parameter);
+
+                if (parameterSymbol is null)
+                {
+                    throw new InvalidOperationException($"The symbol for the parameter '{nameof(parameter)}' could not be found");
+                }
+
+                // Assumption: All parameters in a function method thathave an attribute will have just one (the trigger or input/output binding attribute)
+                AttributeData attributeData = parameterSymbol.GetAttributes().First();
```

Comment:
Customer could write an input converter and use a function parameter like this

```

[Function("products")]
public HttpResponseData Run(
               [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequestData req,
               [InputConverter(typeof(MyCustomConverter))] ProductViewModel productvm)
{
   ....    
}
```

Let's add a test to make sure that use case is not producing unexpected results.

Reply:
Used my own fake attribute for the test case covering this because it's much simpler to set up. In this specific example above, I would have to import/create a bunch of types for the test to pass checks unrelated to source generation.

I think my new test case should be the same as using an attribute like InputConverter b/c neither will be recognized as a BindingAttribute attribute.

Reply:
Sounds good. Any attribute should do the trick.